### ISSUE TITLE
No uploads on phone crash?

### ISSUE THREAD

**Issue by rugk** [westnordost/StreetComplete#324]
Could it be that changes are lost when there are crashes during the survey?

E.g. add an opening_hours tag and crash the phone. (stars: `x+1`) Now, restart and you'll see the quest disappeared (as expected) and the number of stars rose (now without `+n`). However, even some hours or so later (when the OSM server closes the commit) the entry is not added…

Hope you can somehow reproduce it. (I also hope it happened that way…)
v0.11

**Comment by westnordost** [comment]
No, they shouldn't be lost. On solving a quest, the following things happen (in this order):

- quest counter rises by one (n+1)
- the quest answer is saved to local database on the phone
- an automatic upload is triggered (if the auto sync setting is on)

On uploading changes, the following things happen (in this order);

- the quest is uploaded, conflicts are handled. If the conflict is unsolvable, the quest is deleted
- otherwise, the quest is deleted and
- The number of stars rises by one (+n)

The change, if any was made, is visible **immediately** on your OSM history right when the changeset is opened. You can see open changesets.
So, when you get the star, the OSM server said "OK, I accept the change".

So, no idea how you got this, perhaps the one change was added to an older changeset that was still open which you didn't expect it to be and thus did not check? Let me know if you can reproduce it somehow, as said, the situation that you describe should not be possible.
